Description: Pass on the command line parameters to GTK only if not invoked via pkexec
Origin: vendor
Author: Matthias Gerstner <mgerstner@suse.de>
Reviewed-by: Salvatore Bonaccorso <carnil@debian.org>
Last-Update: 2022-01-15

diff --git a/main.c b/main.c
index e0cb7c1..5f7a1e7 100644
--- a/main.c
+++ b/main.c
@@ -7,6 +7,8 @@
 	#include <config.h>
 #endif
 
+#include <stdlib.h>
+
 #include <gtk/gtk.h>
 
 #include "usbtree.h"
@@ -14,8 +16,11 @@
 int main (int argc, char *argv[])
 {
 	GtkWidget *window1;
+	gboolean is_pkexec = getenv("PKEXEC_UID") != NULL;
 
-	gtk_init (&argc, &argv);
+	// only evalute command line parameters if not running in pkexec
+	// privilege escalation context to avoid potential attack vectors
+	gtk_init (is_pkexec ? NULL : &argc, is_pkexec ? NULL : &argv);
 
 	initialize_stuff();
 
